_modules/dotnet-path.test.js

```diff
@@ -0,0 +1,504 @@
+/**************************************************************************************************
+ * Imports
+ **************************************************************************************************/
+
+const faker = require("faker");
+const file  = require("./file");
+const path  = require("path");
+const shell = require("shelljs");
+
+describe("dotnetPath", () => {
+    let dotnetPath;
+    let mockFilePath;
+    beforeEach(() => {
+        mockFilePath = faker.random.word();
+
+        // Due to the way the dotnetPath module caches values, we need to isolate the module in each
+        // test to prevent side effects and test behavior properly.
+        // See https://stackoverflow.com/questions/48989643/how-to-reset-module-imported-between-tests
+        jest.isolateModules(() => {
+            dotnetPath = require("./dotnet-path");
+        });
+    });
+
+    /**************************************************************************************************
+     * cliDir
+     **************************************************************************************************/
+    describe("cliDir", () => {
+        test.each([
+            undefined,
+            null
+        ])("when cliPath() returns %p, it returns undefined", (returnValue) => {
+            // Arrange
+            const cliPathSpy = jest.spyOn(dotnetPath, "cliPath").mockImplementation(() => returnValue);
+
+            // Act
+            const result = dotnetPath.cliDir();
+
+            // Assert
+            expect(cliPathSpy).toHaveBeenCalledTimes(1);
+            expect(result).toBeUndefined();
+        });
+
+        test("when cliPath() returns a value, it calls path.dirname", () => {
+            // Arrange
+            const cliPathSpy = jest.spyOn(dotnetPath, "cliPath").mockImplementation(() => mockFilePath);
+
+            // Act
+            const result = dotnetPath.cliDir();
+
+            // Assert
+            expect(cliPathSpy).toHaveBeenCalledTimes(1);
+            expect(result).toEqual(path.dirname(mockFilePath));
+        });
+    }); // end cliDir
+
+    /**************************************************************************************************
+     * cliPath
+     **************************************************************************************************/
+    describe("cliPath", () => {
+        test.each([
+            undefined,
+            null
+        ])("when solutionDir() returns %p, it returns undefined", (returnValue) => {
+            // Arrange
+            const solutionDirSpy = jest.spyOn(dotnetPath, "solutionDir").mockImplementation(() => returnValue);
+
+            // Act
+            const result = dotnetPath.cliPath();
+
+            // Assert
+            expect(solutionDirSpy).toHaveBeenCalledTimes(1);
+            expect(result).toBeUndefined();
+        });
+
+        test("when solutionDir has a value, it calls file.first with <solutionDir>/Presentation/Cli/bin/Debug/**/*Cli.dll", () => {
```

Comment:
Feel like these paths that are likely internal the the system under test could be exported by that module and simply referenced here with string interpolation for eased refactoring.